tests/safety/common.py

```diff
@@ -219,6 +226,33 @@ def test_steer_req_bit_frames(self):
       self._set_prev_torque(self.MAX_TORQUE)
       self.assertTrue(self._tx(self._torque_cmd_msg(self.MAX_TORQUE, steer_req=1)))
 
+  def test_steer_req_bit_multi_invalid(self):
+    """
+      For safety modes allowing multiple consecutive invalid frames, this ensures that once a valid frame
+      is sent after an invalid frame (even without sending the max number of allowed invalid frames),
+      all counters are reset.
+    """
+    if self.MIN_VALID_STEERING_FRAMES == 0:
+      raise unittest.SkipTest("Safety mode does not implement tolerance for steer request bit safety")
+
+    self.safety.init_tests()
+    self.safety.set_timer(self.MIN_VALID_STEERING_RT_INTERVAL)
+
+    # Allow torque cut
+    self.safety.set_controls_allowed(True)
+    self._set_prev_torque(self.MAX_TORQUE)
+    for _ in range(self.MIN_VALID_STEERING_FRAMES):
+      self.assertTrue(self._tx(self._torque_cmd_msg(self.MAX_TORQUE, steer_req=1)))
+
+    # Send at least 1 or partial amount of allowed invalid frames
+    for _ in range(max(self.MAX_INVALID_STEERING_FRAMES - 1, 1)):
```

Comment:
why can't this just be `self.MAX_INVALID_STEERING_FRAMES`?

Reply:
I wanted this test to ensure that if we don't send the full amount of invalid frames, sending a valid frame again would reset the counters, disallowing oscillating back and forth between valid and invalid, like the comment above talks about

Reply:
Cleaned up to test varying numbers of frames